app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -172,117 +175,136 @@ class EditSetRecordController @Inject() (
     s"Edit Set Record [$editSetRecordId] for Edit Set [$editSetId] was not found"
 
-  private def generateDiscardView(user: User, editSet: EditSet, editSetRecord: EditSetRecord)(implicit
+  private def generateDiscardView(
+    user: User,
+    editSet: EditSet,
+    editSetRecord: EditSetRecord
+  )(implicit
     messages: Messages
-  ): Future[Result] = {
+  ): IO[Result] = IO.pure {
     val title = resolvedMessage(MessageKeys.title)
     val heading = resolvedMessage(MessageKeys.heading, editSetRecord.ccr)
     val message = resolvedMessage(MessageKeys.buttonDiscard)
     val recordType = editSetRecord.recordType
     logger.info(s"Discard changes for record id ${editSetRecord.oci} edit set id ${editSet.id} ")
-    Future.successful(
-      Ok(editSetRecordEditDiscard(user, editSet.name, title, heading, editSetRecord.oci, message, recordType))
-    )
+    Ok(editSetRecordEditDiscard(user, editSet.name, title, heading, editSetRecord.oci, message, recordType))
   }
 
-  private def generateSaveView(user: User, editSet: EditSet, editSetRecord: EditSetRecord)(implicit
+  private def generateSaveView(
+    user: User,
+    editSet: EditSet,
+    editSetRecord: EditSetRecord
+  )(implicit
     messages: Messages
-  ): Future[Result] =
-    Future.successful {
-      val title = resolvedMessage(MessageKeys.title)
-      val heading = resolvedMessage(MessageKeys.heading, editSetRecord.ccr)
-      val message = resolvedMessage(MessageKeys.buttonSave)
-      val recordType = editSetRecord.recordType
-      logger.info(s"Save changes for record id ${editSetRecord.oci} edit set id ${editSet.id}")
-      Ok(editSetRecordEditSave(user, editSet.name, title, heading, editSetRecord.oci, message, recordType))
-    }
+  ): IO[Result] = IO.pure {
+    val title = resolvedMessage(MessageKeys.title)
+    val heading = resolvedMessage(MessageKeys.heading, editSetRecord.ccr)
+    val message = resolvedMessage(MessageKeys.buttonSave)
+    val recordType = editSetRecord.recordType
+    logger.info(s"Save changes for record id ${editSetRecord.oci} edit set id ${editSet.id}")
+    Ok(editSetRecordEditSave(user, editSet.name, title, heading, editSetRecord.oci, message, recordType))
+  }
 
-  private def addAnotherCreator(user: User, editSet: EditSet, editSetRecord: EditSetRecord)(implicit
+  private def addAnotherCreator(
+    user: User,
+    editSet: EditSet,
+    editSetRecord: EditSetRecord
+  )(implicit
     request: Request[AnyContent]
-  ): Result = {
-    val selectedNonEmptyCreatorsFromRequest = filterRequestData { case (key, value) =>
-      key.startsWith(FieldNames.creatorIDs) && value.trim.nonEmpty
+  ): IO[Result] =
+    referenceDataService.getPlacesOfDeposit().map { placesOfDeposit =>
+      val selectedNonEmptyCreatorsFromRequest = filterRequestData { case (key, value) =>
+        key.startsWith(FieldNames.creatorIDs) && value.trim.nonEmpty
+      }
```

Comment:
Should the map be for the entire code block?

Reply:
Correct.

I'll change it.

Reply:
Actually, it is already: I was mistaken.

Reply:
OK